models/scheduled_task.py

```diff
@@ -46,11 +48,11 @@ def __repr__(self):
 
 class ScheduledTaskResult(BaseModel):
     __tablename__ = "scheduled_task_result"
-    id = db.Column(db.Integer, primary_key=True)
-    name = db.Column(db.String, nullable=False)
-    start_time = db.Column(db.DateTime(True), nullable=False)
-    duration = db.Column(db.Interval, nullable=False)
-    result = db.Column(db.JSON, nullable=False)
+    id: Mapped[int] = mapped_column(primary_key=True)
+    name: Mapped[str] = mapped_column()
+    start_time: Mapped[datetime] = mapped_column(DateTime(True))
+    duration: Mapped[timedelta] = mapped_column(Interval)
+    result: Mapped[dict[str, Any]] = mapped_column(JSON)
```

Comment:
Does `Mapped[JSON]` work here?

Reply:
I mean it doesn't cause an error, but I think any uses of `result` are in untyped functions atm, so anything works here. AFAICT `dict[str, Any]` is what this ends up being though?

Reply:
Or are you meaning that it could technically not be a dict and should be typed as any json value?

Can you even do soemthing like 
```python
type JSONValue = None | str | int | float | list[JSONValue] | dict[str, JSONValue]
```

:p

Reply:
Yeah, but I think we should probably be using a custom JSON type that supports mutation anyway, so I'll add that to my list.

Reply:
yeah, even better